line_profiler/__init__.py

```diff
@@ -13,10 +14,17 @@
 
 from .line_profiler import __version__
 
-from .line_profiler import (LineProfiler, LineProfilerMagics,
+from .line_profiler import (LineProfiler,
                             load_ipython_extension, load_stats, main,
                             show_func, show_text,)
 
-__all__ = ['LineProfiler', 'LineProfilerMagics', 'line_profiler',
+__all__ = ['LineProfiler', 'line_profiler',
            'load_ipython_extension', 'load_stats', 'main', 'show_func',
            'show_text', '__version__']
+
+try:
+    from .ipython_extension import LineProfilerMagics
```

Comment:
In the `else` you can simply set `LineProfilerMagics = None` to avoid having to dynamically modify `__all__` (I prefer it when `__all__` is static). We can do the same for the `load_ipython_extension` function, but maybe it becomes a stub function that raises an error if IPython is not available?

Reply:
I changed the way ``LineProfilerMagics`` is defined and added a lazy import. I still think ``LineProfilerMagics`` should be removed from ``__init__.py`` as an implementation detail, but I understand your concern about potential backward compatibility.

Reply:
I would be ok with removing this from the top level API. It doesn't really belong there, and again I doubt anyone is using it. It might be good to search for any potential usage over github and other OSS code repos just in case though. If it doesn't look like it is in use, let's get rid of it if that's easier.